ComponentKit/Core/Scope/CKComponentScopeRootFactory.mm

```diff
@@ -0,0 +1,25 @@
+//
+//  CKComponentScopeRootFactory.m
+//  ComponentKit
+//
+//  Created by Oliver Rickard on 3/28/17.
+//
+//
+
+#import "CKComponentScopeRootFactory.h"
+
+#import "CKComponentControllerAppearanceEvents.h"
+#import "CKComponentBoundsAnimationPredicates.h"
+
+CKComponentScopeRoot *CKComponentScopeRootWithListener(id<CKComponentStateListener> listener)
+{
+  return [CKComponentScopeRoot
+   rootWithListener:listener
+   componentPredicates:{
+     &CKComponentBoundsAnimationPredicate
+   }
+   componentControllerPredicates:{
+     &CKComponentControllerAppearanceEventPredicate,
+     &CKComponentControllerDisappearanceEventPredicate
+   }];
```

Comment:
Makes sense, though this does make it a bit harder to implement more efficient traversals. It may be necessary to also pass along some state that can be used for caching the predicate results. What do you think?